### PR TITLE
Updates go to 1.23.1 & others

### PR DESCRIPTION
**Pull request by JoshVanL**
Updates
- go to 1.23.1
- golangci-lint to 1.61.0
- protoc 5.25.4
- nix flakes lock
- go modules, inc dapr/kit

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,8 +36,8 @@ modtidy:
 # Target: gen-proto                                                            #
 ################################################################################
 PROTOC ?=protoc
-PROTOC_VERSION = 24.4
-PROTOBUF_SUITE_VERSION = 24.4
+PROTOC_VERSION = 25.4
```
Comment (cicoyle) [inline Makefile:39]: Can we keep this the same as dapr/dapr?

Comment (JoshVanL) [reply Makefile:39]: @cicoyle  How about we update the version in dapr instead?

Comment (JoshVanL) [reply Makefile:39]: https://github.com/dapr/dapr/pull/8161

Comment (cicoyle) [reply Makefile:39]: fine by me

**Review by cicoyle — APPROVED** [review]

**Pull request merged**